DNN Platform/Modules/DDRMenu/MenuBase.cs

```diff
@@ -197,22 +197,21 @@ private void FilterNodes(string nodeString, bool exclude)
 			    }
 				else
 				{
-					var nodeText2 = nodeText;
-					filteredNodes.AddRange(
-						RootNode.Children.FindAll(
-							n =>
-							{
-								var nodeName = n.Text.ToLowerInvariant();
-								var nodeId = n.TabId.ToString();
-								return (nodeText2 == nodeName || nodeText2 == nodeId);
-							}));
+                    filteredNodes.Add(RootNode.FindByNameOrId(nodeText));
 				}
 			}
 
             // if filtered for foksonomy tags, use flat tree to get all related pages in nodeselection
 		    if (flattenedNodes.HasChildren())
 		        RootNode = flattenedNodes;
-			RootNode.Children.RemoveAll(n => filteredNodes.Contains(n) == exclude);
+            if (exclude)
+            {
+                RootNode.RemoveAll(filteredNodes);

```

Comment:
I think this condition is reversed. And it is suspicious that this call doesn't actually take the `exclude`  as parameter to pass down to it's children.

Reply:
Actually in older code deletion was not recursive due to which issue was comming for true case i.e for EXCLUDE=true (when EXCLUDE is set) and when false things are working fine for other case